src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java

```diff
@@ -1627,6 +1627,23 @@ void testReconnectWithoutJoiningBefore()
 
 	}
 
+	@Test
+	void testReconnectWithPlayerOnline()
+	{
+		server.addPlayer(player);
+		assertFalse(player.reconnect());
+		assertTrue(server.getOnlinePlayers().contains(player));
+	}
+
+	@Test
+	void testDisconnectWithPlayerOffline()
+	{
+		server.addPlayer(player);
+		assertTrue(player.disconnect());
+		assertFalse(player.disconnect());
+		assertFalse(server.getOnlinePlayers().contains(player));
```

Comment:
This fails since the player is being added to the online player's list twice. perhaps a set should be used and the handling should be cleaned up.